java/src/main/java/com/vmware/OTelConfig.java

```diff
@@ -0,0 +1,62 @@
+package com.vmware;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author Sumit Deo (deosu@vmware.com)
+ */
+public class OTelConfig {
+    private static final String SERVICE_NAME = "otel-otlp-example";
+    public static final String OTEL_COLLECTOR_ENDPOINT = "http://localhost:4317";
+
+
+    /*Adds a BatchSpanProcessor initialized with OtlpGrpcSpanExporter to the
+    TracerSdkProvider.*/
+
+    static OpenTelemetry initOpenTelemetry() {
+        OtlpGrpcSpanExporter spanExporter = getOtlpGrpcSpanExporter();
+        BatchSpanProcessor spanProcessor = getBatchSpanProcessor(spanExporter);
+        Resource serviceNameResource = Resource
+                .create(Attributes.of(ResourceAttributes.SERVICE_NAME, SERVICE_NAME));
+        SdkTracerProvider tracerProvider = getSdkTracerProvider(spanProcessor, serviceNameResource);
+        OpenTelemetrySdk openTelemetrySdk = getOpenTelemetrySdk(tracerProvider);
+        Runtime.getRuntime().addShutdownHook(new Thread(tracerProvider::shutdown));
+
+        return openTelemetrySdk;
+    }
+
+    private static OpenTelemetrySdk getOpenTelemetrySdk(SdkTracerProvider tracerProvider) {
+        OpenTelemetrySdk openTelemetrySdk = OpenTelemetrySdk.builder().setTracerProvider(tracerProvider)
```

Comment:
It might be a bit cleaner to directly return this instead of creating a variable (and on the following functions)

Reply:
Sure, done!!